backend/src/scripts/utility/modelProcessing/checkProcessingFlags.js

```diff
@@ -0,0 +1,103 @@
+/**
+ *  Copyright (C) 2024 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * This script is used to check the processing state of model(s).
+ * Model status should be 'ok' or 'failed' unless it is either 'queued' or 'processing'.
+ * Models that have a 'queued' or 'processing' status while not present in the queued
+ * have in a bad state and resetProcessingFlag can be used to reset it.
+ */
+
+const { v5Path } = require('../../../interop');
+
+const { logger } = require(`${v5Path}/utils/logger`);
+const { getTeamspaceList } = require('../../utils');
+
+const { find } = require(`${v5Path}/handler/db`);
+const { SETTINGS_COL } = require(`${v5Path}/models/modelSettings.constants`);
+const { sendSystemEmail } = require(`${v5Path}/services/mailer`);
+const { templates: emailTemplates } = require(`${v5Path}/services/mailer/mailer.constants`);
+const Path = require('path');
+
+let TIME_LIMIT = 24 * 60 * 60 * 1000; // hours * 1 hour in ms
+
+const processModel = (teamspace, model, status, timestamp) => {
+	logger.logInfo(`\t\t${model} - status: ${status}, timestamp: ${timestamp}`);
+
+	return { teamspace, model, status, timestamp };
+};
+
+const processTeamspace = async (teamspace, model) => {
+	const expiredTimestamp = new Date((new Date()) - TIME_LIMIT);
+	const incompleteStatusQuery = { $or: [{ status: 'processing' }, { status: 'queued' }], timestamp: { $lt: expiredTimestamp } };
```

Comment:
processing statuses is defined in `models/modelSettings.constants`, use that instead of hard coding values.

We should do the reverse so we're covered for any new processing status. Check if the status is not ok, not empty or failed.

Reply:
We need to also look at drawings.

Can you branch from #4830 and also add logic to check for drawings? the status is stored in the revisions instead of model settings.